scripts/parsers/vox-parser.mjs

```diff
@@ -0,0 +1,485 @@
+                for (let x = 0; x <= flattened.extent[0]; ++x) {
+                    pos[0] = x;
+
+                    const v = flattened.at(x, y, z);
+                    const px = flattened.at(x - 1, y, z);
+                    const py = flattened.at(x, y - 1, z);
+                    const pz = flattened.at(x, y, z - 1);
+
+                    if (v !== 0) {
+                        if (px === 0) {
+                            quad(posZ, posY, negX, v);
+                        }
+                        if (py === 0) {
+                            quad(posX, posZ, negY, v);
+                        }
+                        if (pz === 0) {
+                            quad(posY, posX, negZ, v);
+                        }
+                    } else {
+                        if (px !== 0) {
+                            quad(posY, posZ, posX, px);
+                        }
+                        if (py !== 0) {
+                            quad(posZ, posX, posY, py);
+                        }
+                        if (pz !== 0) {
+                            quad(posX, posY, posZ, pz);
+                        }
+                    }
+                }
+            }
+        }
+
+        // construct the mesh
+        const mesh = new pc.Mesh(device);
+        mesh.setPositions(positions);
+        mesh.setNormals(normals);
+        mesh.setColors32(colours);
+        mesh.setIndices(indices);
+        mesh.update();
+
+        return mesh;
+    }
+}
+
+// container resource
+
+class VoxContainerResource {
+    constructor(device, voxModel) {
+        this.device = device;
+        this.voxModel = voxModel;
+    }
+
+    instantiateModelEntity(options) {
+        return null;
+    }
+
+    instantiateRenderEntity(options) {
+        const material = new pc.StandardMaterial();
+        material.diffuseVertexColor = true;
+
+        // generate animation frames
+        const meshInstances = this.voxModel.frames.map((f, i) => {
+            const mesh = VoxGen.mesh(this.device, this.voxModel, i);
+            return new pc.MeshInstance(mesh, material);
+        });
+
+        const entity = new pc.Entity();
+        entity.addComponent('render', {
+            material: material,
+            meshInstances: meshInstances
+        });
+
+        entity.addComponent('voxanim', { });
+
+        this.renders = [];
+
+        return entity;
+    }
+}
+
+// component / system support
+
+const VoxAnimComponentSchema =  ['enabled'];
+
+class VoxAnimComponentData {
+    constructor() {
+        this.enabled = true;
+    }
+}
+
+class VoxAnimComponent extends Component {
+    constructor(system, entity) {
+        super(system, entity);
+
+        this.playing = true;
+        this.timer = 0;
+        this.fps = 10;
+    }
+
+    update(dt) {
+        if (this.playing) {
+            this.timer += dt;
+        }
+
+        const meshInstances = this.entity.render?.meshInstances || this.entity.model?.meshInstances;
```

Comment:
Doesn't the Vox system rely solely on render components now that the model component is legacy?

Reply:
I haven't implemented `VoxContainerResource.instantiateModelEntity()`, but really should since it's part of the public API. Perhaps we can document that model support only exists for GLB container resources instead?

Reply:
Fine by me.